bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -190,56 +337,25 @@ public BookieConnectionPeer getConnectionPeer() {
         }
 
         @Override
-        public void channelBound(ChannelHandlerContext ctx, ChannelStateEvent e) throws Exception {
-            channel = ctx.getChannel();
+        public void channelActive(ChannelHandlerContext ctx) throws Exception {
+            channel = ctx.channel();
```

Comment:
Do we want to call `super.channelRegister(ctx);` too so that it propagates to others in the chain?